modules/blockbuilder/blockbuilder.go

```diff
@@ -232,6 +238,10 @@ func (b *BlockBuilder) running(ctx context.Context) error {
 func (b *BlockBuilder) consume(ctx context.Context) (time.Duration, error) {
 	partitions := b.getAssignedPartitions()
 
+	for _, p := range partitions {
+		metricOwnedPartitions.WithLabelValues(strconv.Itoa(int(p))).Set(1)
```

Comment:
If a partition becomes un-owned, do we have a place to set this back to 0?

Reply:
I may be wrong, but I think't we can leverage on Prometheus staleness. If the metric has not been updated in the past 5 minutes then it won't show up